src_py/apiServer/apiServer.py

```diff
@@ -273,10 +273,13 @@ def download_dataset(self, repo_idx : int | list[int], download_dir_path : str =
             for repo in repo_ids["datasets"]:
                 if repo["idx"] in repo_idx:
                     repo_id = repo["id"]
-                    snapshot_download(repo_id=repo_id, local_dir=f'{download_dir_path}/{repo["name"]}/', repo_type="dataset")
-                    print(f"Files downloaded to {download_dir_path}/{repo['name']}")
+                    full_path_to_repo = f'{download_dir_path}/{repo["name"]}/'
+                    if not os.path.exists(full_path_to_repo):
```

Comment:
@GuyPerets106 
I have added this lines. 
I think that snapshot_download doesn't generate parent directories if they are missing on path.
It seems to solve the issue